camel/models/model_factory.py

```diff
@@ -42,7 +42,10 @@ def create(model_type: ModelType,
         """
         model_class: Any
         if model_type in {
-                ModelType.GPT_3_5_TURBO, ModelType.GPT_4, ModelType.GPT_4_32k
```

Comment:
Notice how you are adding a new model in several places. Ideally it shouldn't be like that. ModelType must be hierarchical so that you only modify the OpenAIModelType by adding a new value to it. Also get_model_token_limit() must be an inherited method of BaseModel. Growing if-elif is what hints that the implementation is not ideal.

Reply:
We now only support OpenAI models and Stub. We can add this hierarchy later. Refactored the get_model_token_limit(). Thanks!

Reply:
Agreed